### PR TITLE
feat(hmr-ws-port): adds ability to explicitly set hmr ws port

### PR DESCRIPTION
**Pull request by Fluidbyte**
This PR is to address issues from #279 in allowing a specific port to be set for the HMR websocket. This allows the use, either in containers, behind firewalls, or in other circumstances where allowing broad port access is not viable, to specifically set the port:

```
-w, --websocket-port <port>    set the port to serve HMR websockets, defaults to random
```

Ensured _non-breaking_ by defaulting to `0` as was previously hard-coded into the WS and HMR configuration.

### CODE REVIEW

File: src/cli.js [hunk 1]
```diff
@@ -13,6 +13,11 @@ program
     'set the port to serve on. defaults to 1234',
     parseInt
   )
+  .option(
+    '-w, --websocket-port <port>',
```
Comment (devongovett) [inline src/cli.js]: can we call the option `--hmr-port` so it's clear what websocket is being referred to?

File: src/HMRServer.js [hunk 2]
```diff
@@ -2,9 +2,9 @@ const WebSocket = require('ws');
 const prettyError = require('./utils/prettyError');
 
 class HMRServer {
-  async start() {
+  async start(websockerPort) {
```
Comment (devongovett) [inline src/HMRServer.js]: misspelled. just call it `port`.

File: src/cli.js (hunk 1 above)
Comment (Fluidbyte) [reply src/cli.js]: yeah, do you want it to be `-h, --hmr-port`?

Comment (devongovett) [reply src/cli.js]: sounds good.

**Pull request merged**